### PR TITLE
fix(components): default table cell padding SCSS variable fix

### LINKED ISSUE #10706: [Bug Report] Default ElTable cell styles cannot be overwritten with SCSS variables
### Element Plus version
2.2.22

### OS/Browsers version
Chrome, 107.0.5304.110 (Official Build) (arm64)

### Vue version
3.2.45

### Reproduction Link
https://github.com/bartoszrudzinski/element-plus-table-padding-bug

### Steps to reproduce
Clone repository, run `npm install` and `npm run dev` in the terminal.

### What is Expected?
Default table cell styles to be overwritten with content of `main.scss` file.

### What is actually happening?
Table styles are not overwritten

### PR DESCRIPTION
**Pull request by bartoszrudzinski**
closed #10706

"default" table cell padding cannot be replaced with $table-cell-padding SCSS variable. This is fixing the issue.

Please make sure these boxes are checked before submitting your PR, thank you!

- [X] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [X] Make sure you are merging your commits to `dev` branch.
- [X] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/table/src/table/defaults.ts [hunk 1]
```diff
@@ -85,9 +85,10 @@ type CellStyle<T> =
       columnIndex: number
     }) => CSSProperties)
 type Layout = 'fixed' | 'auto'
+type Size = 'large' | 'default' | 'small'
 interface TableProps<T> {
   data: T[]
-  size?: string
+  size?: Size
```
Comment (tolking) [inline packages/components/table/src/table/defaults.ts:91]:
```suggestion
  size?: ComponentSize
```

File: packages/components/table/src/table/defaults.ts [hunk 2]
```diff
@@ -179,7 +180,10 @@ export default {
     type: Array as PropType<DefaultRow[]>,
     default: () => [],
   },
-  size: String,
+  size: {
+    type: String as PropType<Size>,
+    default: 'default',
+  },
```
Comment (tolking) [inline packages/components/table/src/table/defaults.ts:186]:
```suggestion
  size: useSizeProp,
```

refer [Button.ts](https://github.com/element-plus/element-plus/blob/dev/packages/components/button/src/button.ts#L26)

**Comment by bartoszrudzinski** [comment]
I updated the code

**Comment by tolking** [comment]
There are some conflicts that need to be resolved.

**Comment by bartoszrudzinski** [comment]
Conflicts resolved

**Review by tolking — APPROVED** [review]

**Review by tolking — APPROVED** [review]

**Review by holazz — APPROVED** [review]

**Pull request merged**